File: euler_2/euler_2.c
#include <stdio.h>

int main()
{
    int sum = 0;
    int prev = 1;
    int fib = 2;
    int next_prev;

    while (fib < 4000000) {
        if (fib % 2 == 0)
            sum += fib;

        next_prev = fib;
        fib += prev;
        prev = next_prev;
    }


    printf("The answer is: %d\n", sum);

    return 0;
}

File: euler_2/euler_2.js
var sum = 0, prev = 1, fib = 2, nextPrev;

while (fib < 4000000) {
    if (fib % 2 === 0) {
        sum += fib;
    }

    nextPrev = fib;
    fib += prev;
    prev = nextPrev;
};

console.log("The answer is: " + sum);
